src/Illuminate/Database/Eloquent/FactoryBuilder.php

```diff
@@ -131,7 +131,22 @@ protected function makeInstance(array $attributes = [])
 
             $definition = call_user_func($this->definitions[$this->class][$this->name], $this->faker, $attributes);
 
-            return new $this->class(array_merge($definition, $attributes));
+            $evaluated = $this->evaluateClosures(array_merge($definition, $attributes));
+
+            return new $this->class($evaluated);
         });
     }
+
+    /**
+     * Delay closures evaluation after merging to avoid duplication.
+     *
+     * @param  array  $attributes
+     * @return array
+     */
+    protected function evaluateClosures(array $attributes)
```

Comment:
does it hurt to have it? (honest question) It may not be necessary but it makes the code more readable.

Devin Dombrowski
dddesigns3@gmail.com




> On Feb 10, 2016, at 7:23 AM, Graham Campbell <notifications@github.com> wrote:
> 
> In src/Illuminate/Database/Eloquent/FactoryBuilder.php <https://github.com/laravel/framework/pull/12231#discussion_r52471200>:
> 
> >          });
> >      }
> > +
> > +    /**
> > +     * Delay closures evaluation after merging to avoid duplication.
> > +     *
> > +     * @param  array  $attributes
> > +     * @return array
> > +     */
> > +    protected function evaluateClosures(array $attributes)
> This function isn't really needed.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/laravel/framework/pull/12231/files#r52471200>.
>